libraries/botbuilder-lg/src/activityFactory.ts

```diff
@@ -0,0 +1,361 @@
+                    break;
+                case 'suggestedactions':
+                    activity.suggestedActions = this.getSuggestions(value);
+                    break;
+                case 'attachmentlayout':
+                    activity.AttachmentLayout = value.toString();
+                default:
+                    console.log(`Skipping unknown activity property ${ property }`);
+                    break;
+            }
+        }
+
+        return activity;
+    }
+
+    private static getSuggestions(value: any): SuggestedActions {
+        let suggestedActions: SuggestedActions = {
+            actions : [],
+            to: []
+        };
+
+        let actions: any[] = this.normalizedToList(value);
+
+        for (const action of actions) {
+            if (typeof action === 'string') {
+                const actionStr: string = action.trim();
+                suggestedActions.actions.push({
+                    type: ActionTypes.MessageBack,
+                    title: actionStr,
+                    displayText: actionStr,
+                    text: actionStr,
+                    value: actionStr
+                });
+            } else {
+                const cardAction: CardAction = this.getCardAction('', action);
+                if (cardAction !== undefined) {
+                    suggestedActions.actions.push(cardAction);
+                }
+            }
+        }
+
+        return suggestedActions;
+    }
+
+    private static getButtons(cardType: string, value: any): CardAction[] {
+        let buttons: any[] = [];
+        let actions: any[] = this.normalizedToList(value);
+
+        for (const action of actions) {
+            if (typeof action === 'string') {
+                const actionStr = action.trim();
+                if (cardType === CardFactory.contentTypes.signinCard ||cardType === CardFactory.contentTypes.oauthCard ) {
+                    buttons.push({
+                        type: ActionTypes.Signin,
+                        title: actionStr,
+                        value: actionStr
+                    });
+                } else {
+                    buttons.push({
+                        type: ActionTypes.ImBack,
+                        title: actionStr,
+                        value: actionStr
+                    });
+                }
+                
+            } else {
+                const cardAction: CardAction = this.getCardAction(cardType, action);
+                if (cardAction !== undefined) {
+                    buttons.push(cardAction);
+                }
+            }
+        }
+
+        return buttons;
+    }
+
+    private static getCardAction(cardType: string, cardActionJObj: any): CardAction {
+        const type: string = this.getStructureType(cardActionJObj);
+        let cardAction: any = {};
+
+        if (cardType === CardFactory.contentTypes.signinCard || cardType === CardFactory.contentTypes.oauthCard) {
```

Comment:
>cardType === CardFactory.contentTypes.signinCard || cardType === CardFactory.contentTypes.oauthCard [](start = 12, length = 99)

Again... This would make every button or suggested action for a signin-card a signin button. That doesn't seem desirable. You can't add a "cancel" or "help" suggested action for instance.